### PR TITLE
Backend - first layout

### LINKED ISSUE #6: Skapa grundstruktur för backend
Skapa grundläggande delar i backend:
1. Environment - nån typ av dependency control. (Antingen requirements.txt eller typ poetry)
2. Grund för hur api:et ska struktureras - dynamiska routes (ev. versioning?)
3. Modul för databashantering? vs använda bibliotek direkt i api:et?
4. Updatera README med installations guide

### PR DESCRIPTION
**Pull request by Leohemmingsson**
# Beskrivning
Jag har skrivit en basic backend i FastAPI, så det finns automatisk documentation (swagger). Backenden kan köras igång med docker-compose, vilket även startar databasen. Exakt hur man startar backend och databas är dokumenterat i README.md

Har även lagt till en fil med ett unittest, men är lite osäker hur man borde på ett bra och relativt smidigt sätt testa funktioner som använder databas. Detta är något som måste kollas upp mer noggrant.

Closes #6

### CODE REVIEW

File: Backend/dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.12
```
Comment (SpiffiRacoon) [inline Backend/dockerfile:1]: Would consider changing the image used for the backend image. The default python docker image i quite large at over 1 GB. Maybe consider using the ```python:3.12-alpine``` image. 

I tested changing to it on my local machine and the swagger documentation as-well as the ping api call still worked. (python-alpine image is 90 MB so it does save considerable space and speed up first time building).

**Review by SpiffiRacoon — CHANGES_REQUESTED** [review]
Code works great, good documentation! (including automatic swagger documentation). However a small change could maybe be made to docker.

**Review by Varlog22 — COMMENTED** [review]
Overall a good start for the backend. The tests that were in place seems to work and the choice of a simplistic UI like Swagger makes it easier to understand what has been implemented.

The commenting seems fine for the level of complexity so far considering the individual functions are quite small in size. As Spiffi said, an Alpine image would minimize the required storage which should be prefered. Great work!

**Review by Varlog22 — CHANGES_REQUESTED** [review]
Overall a good start for the backend. The tests that were in place seems to work and the choice of a simplistic UI like Swagger makes it easier to understand what has been implemented.

The commenting seems fine for the level of complexity so far considering the individual functions are quite small in size. As Spiffi said, an Alpine image would minimize the required storage which should be prefered. Great work!

File: Backend/dockerfile (hunk 1 above)
Comment (Leohemmingsson) [reply Backend/dockerfile]: Thanks, I have updated and tested using the alpine version instead

**Pull request merged**